spring-cloud-gcp-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/core/PubSubTemplate.java

```diff
@@ -65,20 +64,22 @@ public String send(final String topic, Message message) throws RuntimeException
 			try {
 				return Publisher.defaultBuilder(TopicName.create(projectId, topic))
 						.setExecutorProvider(executorProvider).build();
-			}
-			catch (IOException e) {
+			} catch (IOException e) {
 				e.printStackTrace();
 			}
 			return null;
 		});
 
-		try {
-			return publisher.publish(PubsubMessage.newBuilder()
-					.setData(ByteString
-							.copyFrom(message.getPayload().toString().getBytes()))
-					.build()).get();
+		Object pubsubMessageObject = messageConverter.fromMessage(message, PubsubMessage.class);
+
+		if (!(pubsubMessageObject instanceof PubsubMessage)) {
+			throw new MessageConversionException("The specified converter must produce"
+					+ "PubsubMessages to send to Google Cloud Pub/Sub.");
```

Comment:
Oh! Sorry I really told that yesterday. Late work in the evening eliminates any amount of coffee in the morning 😄 

So, I'm waiting for your commit and rebase and merge.
Thanks